crs_motion_planning/src/path_processing_utils.cpp

```diff
@@ -1090,6 +1090,36 @@ bool crs_motion_planning::filterReachabilitySphere(const std::vector<geometry_ms
   return some_points_reachable;
 }
 
+geometry_msgs::msg::PoseArray
+crs_motion_planning::filterSingularityCylinder(const geometry_msgs::msg::PoseArray& waypoints, const double& radius)
```

Comment:
What are these functions used for?

Reply:
There is a potential risk for singularities directly in line with the base cylinder on URs. This function will check for points in that region, as of now it's never being called, but I wanted to preemptively make that function in case we determine it is required.